test/fuzz/lua/mvcc_test.lua

```diff
@@ -0,0 +1,756 @@
+           not ro_txs_mask[stmt.tid] then
+            table.insert(rw_txs, stmt.tid)
+        end
+
+        if stmt.type ~= TXL then
+            table.insert(tx_operations[stmt.tid], stmt)
+        end
+        ::continue::
+    end
+
+-- First of all, try to serialize read-only transactions.
+    for i, tid in ipairs(ro_txs) do
+        if try_apply_tx(tx_operations[tid], false) then ro_txs[i] = nil end
+    end
+
+    for _, rw_tid in ipairs(rw_txs) do
+-- Try to serialize read-write transactions.
+        try_apply_tx(tx_operations[rw_tid], true)
+-- Afterwards, try to serialize read-only transactions again.
+        for i, ro_tid in pairs(ro_txs) do
+            if try_apply_tx(tx_operations[ro_tid], false) then
+                ro_txs[i] = nil
+            end
+        end
+    end
+
+    if next(ro_txs) ~= nil then
+        local failed_ro_set = {}
+        for _, tid in pairs(ro_txs) do
+            table.insert(failed_ro_set, tid)
+        end
+        local msg = 'failed to serialize the following read-only ' ..
+                    'transactions: ' .. table.concat(failed_ro_set, ', ')
+        dump_repro(msg)
+        dump_serial(msg)
+        t.fail(msg)
+    end
+end
+
+local function open_files()
+    local repro_file_name = 'repro.lua'
+    local repro_file_path = fio.pathjoin(g.mvcc.workdir, '..',
+                                         repro_file_name)
+    local err
+    repro_file, err = fio.open(repro_file_path,
+                               {'O_WRONLY', 'O_CREAT', 'O_TRUNC'},
+                               {'S_IRUSR', 'S_IWUSR'})
+    t.fail_if(repro_file == nil, ("`fio.open` failed: %s"):format(err))
+
+    local serial_file_name = 'serial.lua'
+    local serial_file_path = fio.pathjoin(g.oracle.workdir, '..',
+        serial_file_name)
+    serial_file, err = fio.open(serial_file_path,
+                               {'O_WRONLY', 'O_CREAT', 'O_TRUNC'},
+                               {'S_IRUSR', 'S_IWUSR'})
+    t.fail_if(serial_file == nil, ('`fio.open` failed: %s'):format(err))
+end
+
+local function close_files()
+    repro_file:close()
+    serial_file:close()
+end
+
+g.before_all(function()
+    math.randomseed(seed)
+    log.info("Random seed: %d", seed)
+
+    g.cluster = cluster:new{}
+    g.oracle = g.cluster:build_and_add_server{
+        alias   = 'oracle',
+    }
+    g.mvcc = g.cluster:build_and_add_server{
+        alias   = 'mvcc',
+        box_cfg = {
+            memtx_use_mvcc_engine = true
+        }
+    }
+    g.cluster:start()
+    g.oracle:exec(function()
+        require('fiber').set_max_slice({warn = 6000, err = 6000})
+    end)
+    g.mvcc:exec(function()
+        require('fiber').set_max_slice({warn = 6000, err = 6000})
+    end)
+
+    open_files()
+end)
+
+g.after_all(function()
+    close_files()
+    g.cluster:drop()
+end)
+
+g.test_mvcc = function()
+    local schema_creation_code = generate_schema_code()
+
+    while true do
```

Comment:
The test is endless now, I would limit the test by a number of tx's or by time limit.